crates/router/src/connector/iatapay.rs

```diff
@@ -634,55 +632,103 @@ impl api::IncomingWebhook for Iatapay {
-                .parse_struct("IatapayPaymentsResponse")
-                .change_context(errors::ConnectorError::WebhookEventTypeNotFound)?;
-        match notif.status {
-            iatapay::IatapayPaymentStatus::Authorized => match notif.iata_payment_id.is_some() {
-                true => Ok(api::IncomingWebhookEvent::PaymentIntentSuccess),
-                false => Ok(api::IncomingWebhookEvent::RefundSuccess),
-            },
-            iatapay::IatapayPaymentStatus::Failed => match notif.iata_payment_id.is_some() {
-                true => Ok(api::IncomingWebhookEvent::PaymentIntentFailure),
-                false => Ok(api::IncomingWebhookEvent::RefundFailure),
-            },
-            iatapay::IatapayPaymentStatus::Unknown
-            | iatapay::IatapayPaymentStatus::Created
-            | iatapay::IatapayPaymentStatus::Initiated
-            | iatapay::IatapayPaymentStatus::Cleared
-            | iatapay::IatapayPaymentStatus::Settled
-            | iatapay::IatapayPaymentStatus::Tobeinvestigated
-            | iatapay::IatapayPaymentStatus::Blocked
-            | iatapay::IatapayPaymentStatus::Locked
-            | iatapay::IatapayPaymentStatus::UnexpectedSettled => {
-                Ok(api::IncomingWebhookEvent::EventNotSupported)
+        let notif: iatapay::IatapayWebhookResponse = request
+            .body
+            .parse_struct("IatapayWebhookResponse")
+            .change_context(errors::ConnectorError::WebhookEventTypeNotFound)?;
+        match notif {
+            iatapay::IatapayWebhookResponse::IatapayPaymentWebhookBody(wh_body) => {
+                match wh_body.status {
+                    iatapay::IatapayWebhookStatus::Authorized
+                    | iatapay::IatapayWebhookStatus::Settled => {
+                        Ok(api::IncomingWebhookEvent::PaymentIntentSuccess)
+                    }
+                    iatapay::IatapayWebhookStatus::Initiated => {
+                        Ok(api::IncomingWebhookEvent::PaymentIntentProcessing)
+                    }
+                    iatapay::IatapayWebhookStatus::Failed => {
+                        Ok(api::IncomingWebhookEvent::PaymentIntentFailure)
+                    }
+                    iatapay::IatapayWebhookStatus::Created
+                    | iatapay::IatapayWebhookStatus::Cleared
+                    | iatapay::IatapayWebhookStatus::Tobeinvestigated
+                    | iatapay::IatapayWebhookStatus::Blocked
+                    | iatapay::IatapayWebhookStatus::Locked
+                    | iatapay::IatapayWebhookStatus::UnexpectedSettled
+                    | iatapay::IatapayWebhookStatus::Unknown => {
+                        Ok(api::IncomingWebhookEvent::EventNotSupported)
+                    }
+                }
             }
+            iatapay::IatapayWebhookResponse::IatapayRefundWebhookBody(wh_body) => {
+                match wh_body.status {
+                    iatapay::IatapayWebhookStatus::Authorized => {
```

Comment:
refund and payment status seems alike, do we have separate status for both?

Reply:
yes, few status are same!